### ISSUE TITLE
Jdk7Symlink does not delete the test symlink 

### ISSUE THREAD

**Issue by arsi-apli** [gradle/gradle#8119]
In  **org.gradle.internal.nativeintegration.filesystem.jdk7.Jdk7Symlink.doesSystemSupportSymlinks()** is an bug. The created test symbollink is not deleted.

```java
} finally {
            try {
                if (sourceFile != null && sourceFile.toFile().exists()) {
                    Files.delete(sourceFile);
                }
                if (linkFile != null && linkFile.toFile().exists()) {
                    Files.delete(linkFile);
                }
            } catch (IOException e) {
                // We don't really need to handle this.
}
```
The expression  `linkFile.toFile().exists() ` is always false. 
Please change that to:
```java
if (linkFile != null ) {
      Files.deleteIfExists(linkFile);
 }
```

**Comment by marcphilipp** [comment]
@arsi-apli Would you be interested in submitting a PR?

**Comment by marcphilipp** [comment]
Hmm, looks like @DreierF is already working on it. 😉

**Comment by DreierF** [comment]
Seemed related and small enough to fix it in the context of PR #8138 ;-)

**Comment by lacasseio** [comment]
Fixing this issue in a separate PR: https://github.com/gradle/gradle/pull/8519
